.github/workflows/test.yml

```diff
@@ -39,6 +39,8 @@ jobs:
       with:
         path: ~/.cache/pip
         key: pip-test-${{ matrix.python-version }}-${{ matrix.os }}
+    - name: Install setuptools
+      run: pip install setuptools
```

Comment:
If setuptools is a crucial testing dependency, maybe it should instead be added to test dependencies in `pyproject.toml`?